source/puppet/5.3/ssl_autosign.markdown

```diff
@@ -6,75 +6,73 @@ title: "SSL configuration: autosigning certificate requests"
 
 ### Security implications of naïve autosigning
 
-**You should never do this in a production deployment.** Naïve autosigning is only suitable for temporary test deployments that are incapable of serving catalogs containing sensitive information.
+**Never do this in a production deployment.** Naïve autosigning is suitable only for temporary test deployments incapable of serving catalogs containing sensitive information.
 
-Basic Autosigning (autosign.conf)
------
+## Basic autosigning (autosign.conf)
 
 [inpage_basic]: #basic-autosigning-autosignconf
 
 In basic autosigning, the CA uses a config file containing a whitelist of certificate names and domain name globs. When a CSR arrives, the requested certificate name is checked against the whitelist file. If the name is present (or covered by one of the domain name globs), the certificate is autosigned; if not, it is left for manual review.
 
 ### Enabling basic autosigning
 
-To enable basic autosigning, set `autosign = <whitelist file>` in the `[master]` section of the CA Puppet master's puppet.conf. The whitelist file must **not** be executable by the same user as the Puppet master; otherwise it will be treated as a policy executable.
+The `autosign.conf` whitelist file's location and contents are described in [its documentation](./conf_file_autosign.html).
 
-> **Note:** Basic autosigning is enabled by default and looks for a whitelist located at `$confdir/autosign.conf`. For more info, see [the page on the confdir](./dirs_confdir.html).
+Puppet looks for `autosign.conf` at the path configured in the [`autosign` setting][autosign setting] in the `[master]` section of `puppet.conf`. The default path is `$confdir/autosign.conf`, and the default `confdir` path depends on your operating system. [See the confdir documentation for more information.](./dirs_confdir.html)
 
-### The `autosign.conf` file
+> **Note:** In open source Puppet, no `autosign.conf` file exists by default. In Puppet Enterprise, the file exists but is empty. In both cases, basic autosigning is enabled by default but doesn't autosign any certificates because an blank or non-existent whitelist is considered empty and no hostnames are whitelisted by default.
 
-The `autosign.conf` whitelist file is a list of certnames or domain name globs (one per line) whose certificate requests will automatically be signed.
-
-    rebuilt.example.com
-    *.scratch.example.com
-    *.local
-
-Note that domain name globs do not function as normal globs: an asterisk can only represent one or more subdomains at the front of a certname that resembles a fully-qualified domain name. (That is, if your certnames don't look like FQDNs, you can't use `autosign.conf` to full effect.)
+The `autosign.conf` file must not be executable by the Puppet master's user account. If the `autosign` setting points to an executable file, Puppet instead treats it like a custom policy executable even if it contains a valid `autosign.conf` whitelist.
```

Comment:
what happens if it is executable by the Puppet master's user account?

Reply:
> Puppet instead treats it like a custom policy executable even if it contains a valid `autosign.conf` whitelist.

Puppet tries to run it and is probably pretty bummed by the result.

Reply:
I think maybe that nuance would be valuable to include. I didn't get that on my first read. Maybe
"If the `autosign` setting points to an executable file, Puppet instead treats it like a custom policy executable, and attempts to execute it, even if it contains a valid `autosign.conf` whitelist."